tests/test_density.py

```diff
@@ -0,0 +1,170 @@
+    u3, v3 = (grid.dK_dx[time + .5 * dt, lon2, lat2], grid.dK_dy[time + .5 * dt, lon2, lat2])
+    lon3, lat3 = (particle.lon + u3*f_lon, particle.lat + v3*f_lat)
+    u4, v4 = (grid.dK_dx[time + dt, lon3, lat3], grid.dK_dy[time + dt, lon3, lat3])
+    Vx = (u1 + 2*u2 + 2*u3 + u4) / 6.
+    Vy = (v1 + 2*v2 + 2*v3 + v4) / 6.
+    particle.lon += Vx * f_lon * 50000
+    particle.lat += Vy * f_lat * 50000
+
+
+def CreateForcingFields(xdim=200, ydim=350, time=25):
+    depth = np.zeros(1, dtype=np.float32)
+    time = np.arange(0, time+1, time/2, dtype=np.float64)
+
+    # Coordinates of the test grid (on A-grid in deg)
+    lon = np.linspace(0, 1, xdim, dtype=np.float32)
+    lat = np.linspace(1, 2, ydim, dtype=np.float32)
+
+    # Define arrays U (zonal), V (meridional) forcing as zero
+    U = np.zeros((lon.size, lat.size, time.size), dtype=np.float32)
+    V = np.zeros((lon.size, lat.size, time.size), dtype=np.float32)
+
+    ufield = Field('U', U, lon, lat, depth=depth,
+                   time=time, transpose=True)
+    vfield = Field('V', V, lon, lat, depth=depth,
+                   time=time, transpose=True)
+
+    return {'U': ufield, 'V': vfield}
+
+
+def CreateGradientField(grid, mu=None):
+    """Generate a simple gradient field containing a single maxima and gaussian gradient
+    """
+    depth = np.zeros(1, dtype=np.float32)
+    time = grid.time
+
+    lon, lat = grid.U.lon, grid.U.lat
+
+    K = np.zeros((lon.size, lat.size, time.size), dtype=np.float32)
+
+    if mu is None:
+        mu = [np.mean(grid.U.lon), np.mean(grid.U.lat)]
+
+    def MVNorm(x, y, mu=[0, 0], sigma=[[1, 0], [0, 1]]):
+        mu_x = mu[0]
+        mu_y = mu[1]
+        sigma = np.array(sigma)
+        sig_x = sigma[0, 0]
+        sig_y = sigma[1, 1]
+        sig_xy = sigma[1, 0]
+
+        pd = 1/(2 * np.pi * sig_x * sig_y * np.sqrt(1 - sig_xy))
+        pd = pd * np.exp(-1 / (2 * (1 - np.power(sig_xy, 2))) * (
+                        ((np.power(x - mu_x, 2)) / np.power(sig_x, 2)) +
+                        ((np.power(y - mu_y, 2)) / np.power(sig_y, 2)) -
+                        ((2 * sig_xy * (x - mu_x) * (y - mu_y)) / (sig_x * sig_y))))
+
+        return pd
+
+    sig = [[0.3, 0], [0, 0.3]]
+    for i, x in enumerate(lon):
+        for j, y in enumerate(lat):
+            K[i, j, :] = MVNorm(x, y, mu, sig)
+
+    # Boost to provide enough force for our gradient climbers
+    K_Field = Field('K', K, lon, lat, depth, time, transpose=True)
+
+    return K_Field
+
+
+def CreateInitialPositionField(grid):
+    # Simple evenly distribution starting density field
+    depth = grid.U.depth
+    time = grid.U.time
+    lon = grid.U.lon
+    lat = grid.U.lat
+    Num = np.zeros((lon.size, lat.size, time.size), dtype=np.float32)
+
+    for i, x in enumerate(lon):
+        for j, y in enumerate(lat):
+            Num[i, j, :] = 1
+
+    return Field('Start', Num, lon, lat, depth, time, transpose=True)
+
+
+if __name__ == "__main__":
```

Comment:
The test_denisty.py function doesn't seem to be py-tested. Have a look at the other functions in this test-directory, and make sure it gets tested on both 'jit' and 'scipy' mode